src/app/[locale]/rest-client/page.tsx

```diff
@@ -27,115 +27,97 @@ export default function RestClientPage() {
     setError(null);
     setResponse(null);
 
+    if (values.contentType === ContentType.JSON && values.data?.trim()) {
+      const isJsonValid = validateJson(values.data, values.contentType);
+
+      if (!isJsonValid) {
+        setError(ERROR_MESSAGES.INVALID_JSON);
+        return;
+      }
+    }
+
+    const headersArray = Array.isArray(values.headers) ? values.headers : [];
+
+    const invalidHeaders = headersArray.filter(
+      (header: Header) =>
+        (header.key.trim() && !header.value.trim()) || (!header.key.trim() && header.value.trim())
+    );
+
+    if (invalidHeaders.length > 0) {
+      setError(ERROR_MESSAGES.KEY_AND_VALUE);
+      return;
+    }
     try {
-      const response = await axios.post('/api/rest-client', values);
+      const headersObject: Record<string, string> = {};
+
+      headersArray.forEach((header: Header) => {
+        if (header.key.trim() && header.value.trim()) {
+          headersObject[header.key.trim()] = header.value.trim();
+        }
+      });
+
+      const requestData: RequestBody = {
+        method: values.method,
+        url: values.url,
+        headers: headersObject,
+        data: values.data?.trim() || undefined,
+        contentType: values.contentType,
+      };
```

Comment:
If you destructurize as indicated in the comment to the file

```suggestion
      const requestData: RequestBody = {
        method,
        url,
        headers: headersObject,
        data: data?.trim() || undefined,
        contentType,
      };
```

Reply:
Thank you, I have done it like that!